lib/fcm/fcm.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+import '../base/firebase.dart';
+import 'message.dart';
+
+/// FCM Option defined here: https://firebase.google.com/docs/cloud-messaging/send-message#rest
+/// Don't know why you should overwrite these settings but you can.
+class FCMConfig {
+  static const scheme = 'https';
+  static const host = 'fcm.googleapis.com';
+  static const method = 'POST';
+  static const bool keepAlive = false;
+  final headers = <String,String>{};
+  String project_id;
+
+  FCMConfig(this.project_id);
+
+  String get path => '/v1/projects/$project_id/messages:send';
+}
+
+class FCM {
+  /* Singleton instance */
+  static FCM _instance;
+
+  static bool get initialized => _instance != null;
+
+  static FCM initialize(
+      {Firebase firebase, FCMConfig fcmConfig}) {
+    assert(!initialized,
+    'Firestore global instance is already initialized. Do not call this twice or create a local instance via FCM()');
```

Comment:
I know I said copy pasting but I didn't quite mean it literally. This isn't Firestore. :P

Reply:
😄 sorry missed those ones.

Reply:
😄 Sorry missed those ones, fixed now.